server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -59,7 +78,47 @@ public static Ad convertRowToAd(String[] row) {
     return ad; 
   }
 
+  /*
+   * Helper function for cleaning up strings.
+   */
+  public static String trimAndLower(String str) { 
+    return str.trim().toLowerCase(); 
+  }
+
+  /*
+   * Helper function tries to convert date strings to LocalDates to ensure they are 
+   * in isostring format.
+   * checkDateFormat does *not* return the parsed LocalDate because Firestore does not
+   * support that datatype. Dates are converted after retrieval from Firestore.
+   */
+  public static String checkDateFormat(String date) {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    LocalDate localDate = LocalDate.parse(date, formatter);
+    return date;
+  }
+
+  /* 
+   * getImpressions attempts to parse the impressions field for min and max impressions.
+   * Impression field stores impressions in one of the following formats, where n and m are numbers:
+   *      (a) <= n
+   *      (b) n-m
+   * In the case of a, min impressions = 0 and max impressions = n.
+   * In the case of b, min impressions = n and max impressions = m.
+   */ 
+  public static long getImpressions(String impressionsField, String impressionType) throws IllegalArgumentException, NumberFormatException { 
+    String[] impressionsArray = formatImpressionField(impressionsField); 
+    if (impressionType.equals("min")) { // Get minimum number of impressions.
+      if (impressionsArray.length > 1) {
+        return Long.parseLong(impressionsArray[0]);
+      }
+      return (long) 0; // If csv field uses "<=", the min number of impressions is 0.
+    } else { // Get maximum number of impressions.
```

Comment:
you should check whether they put "max" and throw an exception otherwise (or just use an enum) otherwise if I type "nin" instead of "min" you'll return the max.